src/Material/Menu/Oracle.elm

```diff
@@ -0,0 +1,87 @@
+module Material.Menu.Oracle
+  ( Oracle, Element
+  , decode, decode'
+  ) where
+
+import DOM
+import Json.Decode exposing (..)
+
+{-| An Oracle stores relevant information from DOM during Toggle and Close
+events. This computes more than it needs to.
+-}
+
+type alias Oracle =
+  { button : Element
+  , menu : Element
+  , container : Element
+  , offsetTops : List Float
+  , offsetHeights : List Float
+  }
+
+type alias Element =
+  { offsetTop : Float
+  , offsetLeft : Float
+  , offsetHeight : Float
+  , bounds : DOM.Rectangle
+  }
+
+{-| Decode Oracle from the button's reference
+-}
+
+decode : Decoder Oracle
+decode =
+  object5
```

Comment:
I think you can simply do `object5 Oracle`, where `Oracle` is an auto-generated version of the function you define in the next 7 lines.